test/browser-tests/options/options.js

```diff
@@ -20,209 +20,117 @@
 
 describe('Test Options Parameters', function() {
   const swUtils = window.goog.swUtils;
-  const serviceWorkersFolder = '/test/browser-tests/options/serviceworkers';
-
-  const pausePromise = timeout => {
-    return new Promise(function(resolve) {
-      setTimeout(resolve, timeout);
-    });
+  const serviceWorkersFolder = '/test/browser-tests/options/serviceworkers/';
+
+  /**
+   * @param {Number} timeout The number of milliseconds to pause for.
+   * @return {Promise} A promise that resolves after a specified delay.
+   */
+  const pause = timeout => {
+    return new Promise(resolve => setTimeout(resolve, timeout));
   };
 
-  const cleanUpIDB = () => {
-    return new Promise(resolve => {
-      const req = indexedDB.deleteDatabase('sw-toolbox-options-test');
-      req.onsuccess = () => resolve();
-      req.onerror = () => resolve();
-      req.onblocked = () => resolve();
-    });
+  /**
+   * Performs a series of fetch() calls on an iframe, then pauses.
+   * @param {iframe} iframe The iframe whose contentWindow will be used to fetch().
+   * @param {Array.<String>} urls The URLs to fetch.
+   * @return {Promise} A promise that resolves following the fetches and a delay.
+   */
+  const sequentialFetch = (iframe, urls) => {
+    return urls.reduce((chain, url) => {
+      return chain.then(() => iframe.contentWindow.fetch(url));
+    }, Promise.resolve()).then(() => pause(500));
   };
 
-  beforeEach(function() {
-    // Clear IndexDB - Used for max age / max Entries
-    return cleanUpIDB();
+  /**
+   * Prepends a common prefix to several partial URLs, and returns the absolute URLs.
+   * @param {Array.<String>} urls The partial URLs.
+   * @return {Array.<String>} The absolute URLs.
+   */
+  const absoluteTestDataFileUrls = urls => urls.map(url => {
+    return String(new URL(url, `${location.origin}/test/data/files/`));
   });
 
-  after(function() {
-    // Clear IndexDB - Used for max age / max Entries
-    return cleanUpIDB();
-  });
+  /**
+   * Asserts that the keys in cachedAssets match exactly the list of expected URLs.
+   * @param {Object} cachedAssets The result from a call to swUtils.getAllCachedAssets().
+   * @param {Array.<String>} expectedUrls The expected cache contents.
+   */
+  const assertCacheContents = (cachedAssets, expectedUrls) => {
+    const expectedLength = expectedUrls.length;
+    const cachedUrls = Object.keys(cachedAssets);
+    const filteredUrls = cachedUrls.filter(url => expectedUrls.includes(url));
```

Comment:
All of the tests in this file assume that the cached contents need to be exactly equal to the expected contents. I.e., if there's something in cached contents that's not in expected, or vice versa, then it's a failure. This logic seemed like the easiest way to check that bijection.

It's not really a general-purpose check that's applicable to other tests, since other tests might be implemented in such a way that extra entries in the cached contents that aren't expected wouldn't be considered a failure.